event-jq processor and jq variable binding operator ("expression as $variable")

Hi,

I'm attempting to use the event-jq processor, but it doesn't appear to work when the expression includes the variable binding operator "expression as $variable".

The example "test-processor1" below works as expected but "test-processor2" does not and the following error appears in the log.

`[file_output:output1] failed initializing event processor 'test-processor2' of type='event-jq': unexpected token "|"`


#### Config excerpt
```
processors:
  test-processor1:
    event-jq:
      expression: '.[] | .tags.a=1'
      debug: true

  test-processor2:
    event-jq:
      expression: '.[] | .tags.a=1 | .tags.a as $x | .tags.b=$x+1'
      debug: true
```

#### Log output
```
$ ./gnmic --config test_conf.yaml sub --log
2023/07/29 13:12:19.402715 [gnmic] version=0.31.1, commit=0b8789c, date=2023-06-23T14:20:47Z, gitURL=https://github.com/openconfig/gnmic, docs=https://gnmic.openconfig.net
2023/07/29 13:12:19.402744 [gnmic] using config file "test_conf.yaml"
2023/07/29 13:12:19.402879 [config] validating processor "test-processor1" config
2023/07/29 13:12:19.402896 [config] validating processor "test-processor2" config
2023/07/29 13:12:19.403248 [gnmic] starting output type file
2023/07/29 13:12:19.403309 [gnmic] queuing target "t-bng1"
2023/07/29 13:12:19.403335 [gnmic] starting target "t-bng1" listener
2023/07/29 13:12:19.403457 [gnmic] subscribing to target: "t-bng1"
2023/07/29 13:12:19.403708 [file_output:output1] added event processor 'test-processor1' of type=event-jq to file output
2023/07/29 13:12:19.403809 [file_output:output1] failed initializing event processor 'test-processor2' of type='event-jq': unexpected token "|"
```

If I test the same expression with the gojq CLI client, it works as expected.

```
$ gojq -v
gojq 0.12.6 (rev: HEAD/go1.17.3)
$ cat test.json | gojq '.[] | .tags.a=1 | .tags.a as $x | .tags.b=$x+1'
```

Assuming my config file is ok, I assume it is failing at the point below (I'm not familiar with Go).

https://github.com/openconfig/gnmic/blob/main/formatters/event_jq/event_jq.go#L70

Regards
Brad

Hi,

Thanks for the quick response. I just retested on 0.31.6 and unfortunately I still get the same "failed initializing event processor" message. As soon as I comment out the event processor under the output section the error disappears.

```
$ ./gnmic version
version : 0.31.6
 commit : dce3d0f
   date : 2023-08-01T00:00:26Z
 gitURL : https://github.com/openconfig/gnmic
   docs : https://gnmic.openconfig.net
$ ./gnmic --config test_conf.yaml sub --log
2023/08/02 20:11:13.390797 [gnmic] version=0.31.6, commit=dce3d0f, date=2023-08-01T00:00:26Z, gitURL=https://github.com/openconfig/gnmic, docs=https://gnmic.openconfig.net
2023/08/02 20:11:13.390825 [gnmic] using config file "test_conf.yaml"
2023/08/02 20:11:13.390942 [config] validating processor "test-processor2" config
2023/08/02 20:11:13.391281 [gnmic] starting output type file
2023/08/02 20:11:13.391340 [gnmic] queuing target "t-bng1"
2023/08/02 20:11:13.391379 [gnmic] starting target "t-bng1" listener
2023/08/02 20:11:13.391428 [gnmic] subscribing to target: "t-bng1"
2023/08/02 20:11:13.391735 [file_output:output1] failed initializing event processor 'test-processor2' of type='event-jq': unexpected token "|"
```
__Conf Excerpt__ 

```
processors:
  test-processor2:
    event-jq:
      expression: '.[] | .tags.a=1 | .tags.a as $x | .tags.b=$x+1'
      debug: true

outputs:
  output1:
    type: file
    filename: /home/abc/gnmi/test.log
    format: event
    debug: true
    event-processors:
      - test-processor2
```

Regards
Brad
P.S. I notice that if I specify "filename: ~/gnmi/test.log" rather than give an absolute path, like above, I get a SIGSEV (using Red Hat). I'm happy to raise a separate issue if you would like me to report this. 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x905d30]

goroutine 28 [running]:
golang.org/x/sync/semaphore.(*Weighted).Acquire(0x0, {0x4642ee8, 0xc000a90640}, 0x1)
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.3.0/semaphore/semaphore.go:41 +0x50
github.com/openconfig/gnmic/outputs/file.(*File).Write(0xc0005e8dc0, {0x4642ee8?, 0xc000a90640?}, {0x4611ac0?, 0xc000050eb0}, 0x0?)
        /home/runner/work/gnmic/gnmic/outputs/file/file_output.go:217 +0x9a
github.com/openconfig/gnmic/app.(*App).Export.func1({0x4658ab0?, 0xc0005e8dc0?})
        /home/runner/work/gnmic/gnmic/app/collector.go:158 +0x12d
created by github.com/openconfig/gnmic/app.(*App).Export
        /home/runner/work/gnmic/gnmic/app/collector.go:154 +0x3f4
```

--- a/config/environment.go
+++ b/config/environment.go
@@ -66,7 +66,7 @@ OUTER:
 			}
 			m[f] = os.ExpandEnv(v)
 		case map[string]interface{}:
-			expandMapEnv(v)
+			expandMapEnv(v, except...)
 			m[f] = v
 		}
 	}

